nemo/collections/nlp/data/language_modeling/megatron/retro_dataset.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""RETRO Style dataset."""
+
+import torch
+
+from nemo.collections.nlp.data.language_modeling.megatron.base_dataset_utils import get_train_valid_test_split_
+from nemo.collections.nlp.data.language_modeling.megatron.megatron_dataset import MegatronDataset
+from nemo.utils import logging
+
+# from nemo.collections.nlp.data.language_modeling.megatron.dataset_utils import (
+#     create_masked_lm_predictions,
+#     create_tokens_and_tokentypes,
+#     get_a_and_b_segments,
+#     get_samples_mapping,
+#     truncate_segments,
+# )
+# from nemo.collections.nlp.data.language_modeling.megatron.indexed_dataset import MMapIndexedDataset
+try:
+    from apex.transformer import parallel_state
+
+    HAVE_APEX = True
+except (ImportError, ModuleNotFoundError):
+    HAVE_APEX = False
+
+
+class MockRETRODataset(MegatronDataset):
+    def __init__(self, cfg, trainer, tokenizer, name, size):
+        super().__init__(cfg, trainer=trainer)
+        self.name = name
+        self.tokenizer = tokenizer
+        self._cfg = cfg
+        self.size = size
+        seed_val = parallel_state.get_data_parallel_rank() * 131 + 97
```

Comment:
seed is not configurable?

Reply:
This is mock dataset. the seed number is used to model different batch for different worker. No need to configure this.